File: 1002-maximum-width-ramp/1002-maximum-width-ramp.cpp
class Solution {
public:
    int maxWidthRamp(vector<int>& nums) {

        int ans= 0;
        int curr = 0;
        vector<int> minis; minis.push_back(0);
        for(int i  =1; i< nums.size(); i++)
        {
            
            int j  = minis.size()-1;
            if(nums[i] < nums[minis[j]])
            {
                minis.push_back(i);
                continue;
            }
            while(j>=0&&nums[i] >= nums[minis[j]])
            {
                ans = max(ans, i-minis[j]);
                j--;
            }
            
        }return ans;
        
    }
};

File: 1002-maximum-width-ramp/1002-maximum-width-ramp.cs
public class Solution {
    public int MaxWidthRamp(int[] nums) {

        int ans= 0;
        int curr = 0;
        int sz = nums.Length;
        List<int> minis = new List<int>(); minis.Add(0);
        for(int i  =1; i< sz; i++)
        {
            
            int j  = minis.Count-1;
            if(nums[i] < nums[minis[j]])
            {
                minis.Add(i);
                continue;
            }
            while(j>=0&&nums[i] >= nums[minis[j]])
            {
                ans = int.Max(ans, i-minis[j]);
                j--;
            }
            
        }return ans;

        
    }
}